Makefile

```diff
@@ -149,7 +149,7 @@ TUTORIAL_CXX_FLAGS ?= -std=c++11 -g -fno-omit-frame-pointer -fno-rtti -I $(ROOT_
 # The tutorials contain example code with warnings that we don't want
 # to be flagged as errors, so the test flags are the tutorial flags
 # plus our warning flags.
-TEST_CXX_FLAGS ?= $(TUTORIAL_CXX_FLAGS) $(CXX_WARNING_FLAGS)
+TEST_CXX_FLAGS ?= $(TUTORIAL_CXX_FLAGS) $(CXX_WARNING_FLAGS) -march=native
```

Comment:
Why is `-march=native` necessary?

Reply:
It sniffs the level of processor one is on and enables all the ISA extensions that currently exist. Without it, AVX2 is not enabled so far as I can tell even if the current host supports it. If one is shipping binaries from one machine to another, that is probably a good thing, but for our test setup, enabling everything seems useful. But to be honest, I'm not 100% sure about this.

Reply:
Probably good to capture some of that in a comment in the Makefile.